### PR TITLE
wait for tenant to be active before polling for timeline absence

### PR DESCRIPTION
**Pull request by LizardWizzard**
## Problem

https://neon-github-public-dev.s3.amazonaws.com/reports/main/5692829577/index.html#suites/f588e0a787c49e67b29490359c589fae/4c50937643d68a66

## Summary of changes

wait for tenant to be active after restart  before polling for timeline absence

### CODE REVIEW

File: test_runner/regress/test_timeline_delete.py [hunk 1]
```diff
@@ -258,12 +258,12 @@ def test_delete_timeline_exercise_crash_safety_failpoints(
     if check is Check.RETRY_WITH_RESTART:
         env.pageserver.stop()
         env.pageserver.start()
-        if failpoint == "timeline-delete-before-index-deleted-at":
-            # We crashed before persisting this to remote storage, need to retry delete request
 
-            # Wait till tenant is loaded. Shouldnt take longer than 2 seconds (we shouldnt block tenant loading)
-            wait_until_tenant_active(ps_http, env.initial_tenant, iterations=2)
+        # Wait till tenant is loaded. Shouldnt take longer than 2 seconds
+        wait_until_tenant_active(ps_http, env.initial_tenant, iterations=2)
```
Comment (koivunej) [inline test_runner/regress/test_timeline_delete.py:263]: If I am not misunderstanding the logs, it took more than `2 < x < 3` seconds in the test for the tenant to finish loading, unsure why. Should we allow for a longer timeout here, or just go with the default. It was with real_s3 after all..

**Review by koivunej — APPROVED** [review]
Changes look good, but wondering about the timeout being enough.

**Comment by koivunej** [comment]
> * `test_concurrent_timeline_delete_stuck_on[in_progress_delete]`: [debug](https://neon-github-public-dev.s3.amazonaws.com/reports/pr-4856/5724673106/index.html#suites/f588e0a787c49e67b29490359c589fae/d5c1e2bef40f7e02)

Re-running.

**Comment by koivunej** [comment]
Re: timeouts, adding this one here as well: https://neon-github-public-dev.s3.amazonaws.com/reports/pr-4857/5718388045/index.html#suites/f588e0a787c49e67b29490359c589fae/5ef37c7fda6cfb7f

TL;DR: another tenant load not completing in 2s, this time aiming for Broken. I think we should raise all timeouts to 10 seconds on tests which are also ran against real_s3.

**Comment by LizardWizzard** [comment]
I'll rerun workflow for several times to be sure that the fix helped

File: test_runner/regress/test_timeline_delete.py [hunk 2]
```diff
@@ -229,6 +229,8 @@ def test_delete_timeline_exercise_crash_safety_failpoints(
 
     ps_http.configure_failpoints((failpoint, "return"))
 
+    iterations = 20 if remote_storage_kind is RemoteStorageKind.REAL_S3 else 4
```
Comment (koivunej) [inline test_runner/regress/test_timeline_delete.py:232]: Do we need to have sometimes faster timeouts, as in, will the quicker response ever be anything except flakyness?

It might be a slow failure if code is changed, but assuming that the tests work and the code works, then larger timeout => more stable test?

Comment (LizardWizzard) [reply test_runner/regress/test_timeline_delete.py:232]: Pasting from DMs:

> I'm still in the "fail fast" camp. This timeline test is parametrized which multiplies potential delay by the factor of 32. If we needlessly wait for 30 seconds instead of 10, this costs 20*32=640 seconds which is slightly above 10 minutes. To me it feels very important to keep test runs fast. They are not fast already, and timeouts make them even slower.
I wont die on that hill though, if you want me to set one big timeout I'll do that

Comment (koivunej) [reply test_runner/regress/test_timeline_delete.py:232]: Yeah, well, this PR is improvement regardless so let's get it in.

Comment (LizardWizzard) [reply test_runner/regress/test_timeline_delete.py:232]: Discussed in DM's, ok to resolve

File: test_runner/regress/test_timeline_delete.py (hunk 1 above)
Comment (LizardWizzard) [reply test_runner/regress/test_timeline_delete.py]: Done
